### PR TITLE
test(line-ripple): Update tests

### PR DESCRIPTION
**Pull request by williamernest**
Add missing tests for line-ripple.

### CODE REVIEW

File: test/unit/mdc-line-ripple/mdc-line-ripple-foundation.test.js [hunk 1]
```diff
@@ -56,20 +56,62 @@ test('#destroy removes event listeners', () => {
...
   const {foundation, mockAdapter} = setupTest();
   foundation.init();
   foundation.activate();
   td.verify(mockAdapter.addClass(cssClasses.LINE_RIPPLE_ACTIVE));
 });
 
-test(`deactivate removes ${MDCLineRippleFoundation.cssClasses.LINE_RIPPLE_DEACTIVATING} class`, () => {
+test(`deactivate adds ${cssClasses.LINE_RIPPLE_DEACTIVATING} class`, () => {
   const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(cssClasses.LINE_RIPPLE_DEACTIVATING)).thenReturn(true);
```
Comment (moog16) [inline test/unit/mdc-line-ripple/mdc-line-ripple-foundation.test.js]: Is this needed? Was it failing before having this?

File: test/unit/mdc-line-ripple/mdc-line-ripple-foundation.test.js [hunk 2]
```diff
@@ -56,20 +56,62 @@ test('#destroy removes event listeners', () => {
...
+
   foundation.init();
   foundation.deactivate();
   td.verify(mockAdapter.addClass(cssClasses.LINE_RIPPLE_DEACTIVATING));
 });
 
+test('opacity event after adding deactivating class triggers activation classes to be removed', () => {
```
Comment (moog16) [inline test/unit/mdc-line-ripple/mdc-line-ripple-foundation.test.js]: Is it more correct to say, "...triggers removal of activation classes"?

File: test/unit/mdc-line-ripple/mdc-line-ripple-foundation.test.js [hunk 3]
```diff
@@ -56,20 +56,62 @@ test('#destroy removes event listeners', () => {
...
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(cssClasses.LINE_RIPPLE_DEACTIVATING)).thenReturn(true);
+  const event = {
+    propertyName: 'opacity',
+  };
+
+  foundation.init();
+  foundation.deactivate();
+  td.verify(mockAdapter.addClass(cssClasses.LINE_RIPPLE_DEACTIVATING));
```
Comment (moog16) [inline test/unit/mdc-line-ripple/mdc-line-ripple-foundation.test.js]: This test seems like 2 in 1. 

One test can just be, "#deactivate adds the cssClasses.LINE_RIPPLE_DEACTIVATING class". The other can be opacity event triggers ...

File: test/unit/mdc-line-ripple/mdc-line-ripple-foundation.test.js [hunk 4]
```diff
@@ -56,20 +56,62 @@ test('#destroy removes event listeners', () => {
...
+  foundation.handleTransitionEnd(event);
+  td.verify(mockAdapter.removeClass(cssClasses.LINE_RIPPLE_DEACTIVATING));
+  td.verify(mockAdapter.removeClass(cssClasses.LINE_RIPPLE_ACTIVE));
+});
+
+test(`non opacity transition event doesn't remove ${cssClasses.LINE_RIPPLE_DEACTIVATING} class`, () => {
+  const {foundation, mockAdapter} = setupTest();
+  td.when(mockAdapter.hasClass(cssClasses.LINE_RIPPLE_DEACTIVATING)).thenReturn(true);
+  const event = {
+    propertyName: 'not-opacity',
```
Comment (moog16) [inline test/unit/mdc-line-ripple/mdc-line-ripple-foundation.test.js]: great event name ;)

File: test/unit/mdc-line-ripple/mdc-line-ripple-foundation.test.js (hunk 2 above)
Comment (williamernest) [reply test/unit/mdc-line-ripple/mdc-line-ripple-foundation.test.js]: No, the line-ripple adds a class that triggers the deactivation, and then when the transition event is finished it removes the active and deactivating classes. 

There's a different animation for idle -> focused then there is for focused -> idle.

**Pull request merged**